modules/blazingsql/wrapper/blazingsql/api.cpp

```diff
@@ -13,39 +13,67 @@
 // limitations under the License.
 
 #include "api.hpp"
+#include "ucpcontext.hpp"
 
 #include <engine/engine.h>
 #include <engine/initialize.h>
 
 namespace nv {
 
 ContextWrapper::wrapper_t initialize(Napi::Env const& env, NapiToCPP::Object const& props) {
-  uint16_t ral_id                               = props.Get("ralId");
-  std::string worker_id                         = props.Get("workerId");
-  std::string network_iface_name                = props.Get("network_iface_name");
-  int32_t ral_communication_port                = props.Get("ralCommunicationPort");
-  std::vector<NodeMetaDataUCP> workers_ucp_info = props.Get("workersUcpInfo");
-  bool single_node                              = props.Get("singleNode");
-  std::string allocation_mode                   = props.Get("allocationMode");
-  std::size_t initial_pool_size                 = props.Get("initialPoolSize");
-  std::size_t maximum_pool_size                 = props.Get("maximumPoolSize");
-  bool enable_logging                           = props.Get("enableLogging");
+  uint16_t ral_id                = props.Get("ralId");
+  std::string worker_id          = props.Get("workerId");
+  std::string network_iface_name = props.Get("networkIfaceName");
+  int32_t ral_communication_port = props.Get("ralCommunicationPort");
+  bool single_node               = props.Get("singleNode");
+  std::string allocation_mode    = props.Get("allocationMode");
+  std::size_t initial_pool_size  = props.Get("initialPoolSize");
+  std::size_t maximum_pool_size  = props.Get("maximumPoolSize");
+  bool enable_logging            = props.Get("enableLogging");
 
   auto config_options = [&] {
     std::map<std::string, std::string> config{};
     auto prop = props.Get("configOptions");
-    if (prop.IsObject() and not prop.IsNull()) {
+    if (!prop.IsNull() && prop.IsObject()) {
       auto opts = prop.As<Napi::Object>();
       auto keys = opts.GetPropertyNames();
       for (auto i = 0u; i < keys.Length(); ++i) {
         Napi::HandleScope scope(env);
-        auto name    = keys.Get(i).ToString();
-        config[name] = opts.Get(name).ToString();
+        std::string name = keys.Get(i).ToString();
+        config[name]     = opts.Get(name).ToString();
+        if (config[name] == "true") {
+          config[name] = "True";
+        } else if (config[name] == "false") {
+          config[name] = "False";
+        }
       }
     }
     return config;
   }();
 
+  Napi::Array objects               = props.Get("workersUcpInfo");
+  UcpContext::wrapper_t ucp_context = UcpContext::wrapper_t();
```

Comment:
I think the only thing I'd need a suggestion on how to handle, we create a `ucp_context` for all cases, should we avoid creating it if we are doing only single node?

Reply:
This is constructing an empty `Napi::Object` instance, which should be fine as long as we don't try to use it when `singleNode=true`.

Reply:
ah right, should be fine then.